provider/bluecat/bluecat.go

```diff
@@ -219,11 +235,27 @@ func (p *BluecatProvider) Records(ctx context.Context) (endpoints []*endpoint.En
 
 				for _, ip := range ips {
 					ep := endpoint.NewEndpointWithTTL(propMap["absoluteName"], endpoint.RecordTypeA, endpoint.TTL(ttl), ip)
+					for _, txtRec := range resT {
+						if strings.Compare(rec.Name, txtRec.Name) == 0 {
+							ep.Labels[endpoint.OwnerLabelKey], err = extractOwnerfromTXTRecord(txtRec.Properties)
+							if err != nil {
+								return nil, errors.Wrapf(err, "owner not parsed correctly")
+							}
+						}
+					}
 					endpoints = append(endpoints, ep)
 				}
 			} else {
 				for _, ip := range ips {
 					ep := endpoint.NewEndpoint(propMap["absoluteName"], endpoint.RecordTypeA, ip)
+					for _, txtRec := range resT {
+						if strings.Compare(rec.Name, txtRec.Name) == 0 {
+							ep.Labels[endpoint.OwnerLabelKey], err = extractOwnerfromTXTRecord(txtRec.Properties)
+							if err != nil {
```

Comment:
Could we condense this logic with the code on lines 238-245 in this PR? Maybe add a conditional regarding the record TTL inside a single for loop (rather than outside of 2 for loops that essentially do the same thing).